dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/PublicUserFilter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.BeanPropertyWriter;
+import com.fasterxml.jackson.databind.ser.PropertyWriter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import io.dockstore.webservice.core.AuthenticatedUser;
+import io.dockstore.webservice.core.User;
+
+/**
+ * A Jackson JSON filter that writes out false for the User#isAdmin and User#curator properties, if the requesting user is not an admin and does not
+ * match the user making the request.
+ */
+public class PublicUserFilter extends SimpleBeanPropertyFilter {
+
+    protected static final String ADMIN_PROPERTY = "isAdmin";
+    protected static final String CURATOR_PROPERTY = "curator";
+    private BeanPropertyWriter maskCuratorAndAdminWriter = new MaskAdminCuratorPropertyWriter();
+
+
+    @Override
+    public void serializeAsField(Object pojo, JsonGenerator jgen, SerializerProvider provider, PropertyWriter writer) throws Exception {
+        User user = (User)pojo;
```

Comment:
This filter is currently only used for a User because of [here](https://github.com/dockstore/dockstore/blob/a97caabcba07b959776be15cef0fd592adea0fe0/dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java#L361) and [here](https://github.com/dockstore/dockstore/blob/c5eeb07ed8dbc32dfee1968ca4f4b1f19799c4e6/dockstore-webservice/src/main/java/io/dockstore/webservice/core/User.java#L102). But it's not enforced with the compiler, so I'll add an instance check as well to be safe.

Reply:
Added an instance check.